workers/unity/Assets/Playground/Scripts/Metrics/MetricSendSystem.cs

```diff
@@ -0,0 +1,75 @@
+using System.Collections.Generic;
+using Unity.Entities;
+using UnityEngine;
+using Improbable.Gdk.Core;
+
+namespace Playground
+{
+    public class MetricSendSystem : ComponentSystem
+    {
+        private WorkerBase worker;
+
+        private float timeElapsedSinceUpdate = 0.0f;
+
+        private List<float> fpsMeasurements = new List<float>();
+
+        protected override void OnCreateManager(int capacity)
+        {
+            base.OnCreateManager(capacity);
+            worker = WorkerRegistry.GetWorkerForWorld(World);
+        }
+
+        protected override void OnUpdate()
+        {
+            if (worker.Connection == null)
+            {
+                return;
+            }
+
+            var connection = worker.Connection;
+
+            timeElapsedSinceUpdate += Time.deltaTime;
+            AddFpsSample();
+            if (timeElapsedSinceUpdate >= MetricConfig.TimeBetweenMetricUpdatesSecs)
+            {
+                timeElapsedSinceUpdate = 0;
+                float fps = CalculateFps();
+                var load = MetricConfig.CalculateLoad == null ?
+                    DefaultLoadCalculation(fps) : MetricConfig.CalculateLoad(fps);
+                Improbable.Worker.Metrics metrics = new Improbable.Worker.Metrics
+                {
+                    Load = load
+                };
+                connection.SendMetrics(metrics);
+            }
+        }
+
+        private float DefaultLoadCalculation(float fps)
+        {
+            float targetFps = Application.targetFrameRate;
+            return Mathf.Max(0.0f, (targetFps - fps) / (0.5f * targetFps));
+        }
+
+        private void AddFpsSample()
+        {
+            if (fpsMeasurements.Count == MetricConfig.MaxFpsSamples)
+            {
+                fpsMeasurements.RemoveAt(0);
```

Comment:
This seems a little odd. If you specify a sample rate then surely a maximum can't be hit, in fact you wouldn't be guaranteed to hit the expected minimum. list.RemoveAt(0) always make uncomfortable.

Reply:
Could we also just accumulate the total fps can keep track of the count. Or do we need the intermediate values?

Reply:
There isn't really a 'sample rate'. This is a 'sliding window' approach to computing the mean. I agree that list.RemoveAt(0) isn't great -- I guess I could use a queue instead.

Reply:
I'd suggest changing that anyway. Without a constrained frame rate this will just take a large number of samples near the beginning of the tick then max out.

Reply:
I'm not entirely sure I understand the comment? The FPS estimate should rely on the more recent measurements. An alternative would be to use exponential averaging.